### PR TITLE
Mechanism to link colors across visualization #152

### PR DESCRIPTION
**Pull request by nxjiten**
@natabal, @killanch

Add a mechanism to map colors across visualizations. To use this mapping colors we have to define an object of `value => color` format in visualization settings E.g - 

`
"mappedColors": {
            "WEBEX": "#D9070A",
            "NETFLIX": "#FFBF02",
            "CISCOVPN": "#62AC00"
        }
`

Please have a look and let me know.

### CODE REVIEW

**Comment by natabal** [comment]
@nxjiten We cannot add those in the settings as those are runtimes values: WEBEX, NETFLIX, etc.

**Comment by nxjiten** [comment]
@natabal 
`_@nxjiten We cannot add those in the settings as those are runtimes values: WEBEX, NETFLIX, etc._`

Ok, I understand. I will make changes according.

**Comment by nxjiten** [comment]
@natabal 
Add color mapping for runtime values.
Dependent PR - https://github.com/nuagenetworks/vsd-react-ui/pull/2815

File: Graphs/AbstractGraph.js [hunk 1]
```diff
@@ -260,8 +264,9 @@ export default class AbstractGraph extends React.Component {
             return;
 
         const scale = d3.scaleOrdinal([...colors, ...otherColors]);
-        scale.domain(data.map((d) => d[colorColumn || defaultColumn]));
+        const domainData = d3.map(data, (d) => d[colorColumn || defaultColumn]).keys().sort();
 
+        scale.domain((setGraphColor && setGraphColor(domainData)) || domainData);
```
Comment (natabal) [inline Graphs/AbstractGraph.js]: This is great it gives a the same color scheme however what I think we are looking for is a mapping between runtime values and colors, as per your example: `{ "WEBEX": "#D9070A", "NETFLIX": "#FFBF02", "CISCOVPN": "#62AC00" }`

**Pull request merged**